AmsMigrator/Infrastructure/OkapiClient.cs

```
@@ -0,0 +1,228 @@
+        };
+
+        public OkapiClient(ImportOptions options)
+        {
+            _httpClient = new HttpClient();
+            _httpClient.Timeout = TimeSpan.FromSeconds(30);
+            _httpClient.SetToken(options.OkApiAuthToken);
+            _httpClient.DefaultRequestHeaders.ConnectionClose = true;
+
+            _baseUri = options.OkApiBaseUri;
+            _apiVersion = options.OkApiApiVersion;
+
+            _retryPolicy = Policy
+              .Handle<HttpRequestException>()
+              .Or<TaskCanceledException>()
+              .OrResult<HttpResponseMessage>(r => _httpStatusCodesWorthRetrying.Contains(r.StatusCode))
+              .WaitAndRetryAsync(options.HttpServicesRetryCount, retryAttempt => TimeSpan.FromSeconds(10), (dr, ts, rc, _) =>
+              {
+                  if (dr.Exception != null)
+                      _logger.Error("Exception {0} occured, retry attempt: {1}, timeout: {2}", dr.Exception.GetType().Name, rc, ts);
+                  if (dr.Result != null)
+                      _logger.Error("Invalid result {0} obtained, retry attempt: {1}, timeout: {2}", dr.Result, rc, ts);
+              });
+        }
+
+        public async Task<MaterialStub> CreateMaterialStubAsync(string type, string code, long firm, string language)
+        {
+            string uri = _baseUri + $"/api/{_apiVersion}/{type}/{code}/session?firm={firm}&languages={language}";
+
+            try
+            {
+                using (var response = await _retryPolicy.ExecuteAsync(() => _httpClient.PostAsync(uri, new StringContent(String.Empty, Encoding.UTF8, "application/json"))))
+                {
+                    var content = await response.Content.ReadAsStringAsync();
+
+                    if (!response.IsSuccessStatusCode)
+                    {
+                        _logger.Error("Unsuccessfull response with code {code} and content {content}", response.StatusCode, content);
+                        response.EnsureSuccessStatusCode();
```

You've already checked `IsSuccessStatusCode` property. Do you need to throw an exception here?